File: leetcode/src/test/java/problems/no2/SolutionTest.java
package problems.no2;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import problems.no2.Solution.ListNode;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

class SolutionTest {

    private final Solution solution = new Solution();
    @Test
    void 테스트_케이스_1() {
        ListNode list1 = ListNode.builder().add(2).add(4).add(3).build();
        ListNode list2 = ListNode.builder().add(5).add(6).add(7).build();

        ListNode expected = ListNode.builder().add(7).add(0).add(8).build();
        ListNode result = solution.addTwoNumbers(list1, list2);

        assertThat(expected, is(equalTo(result)));
    }

    @Test
    void 테스트_케이스_2() {
        ListNode list1 = ListNode.builder().add(0).build();
        ListNode list2 = ListNode.builder().add(0).build();

        ListNode expected = ListNode.builder().add(0).build();
        ListNode result = solution.addTwoNumbers(list1, list2);

        assertThat(expected, is(equalTo(result)));
    }

    @Test
    void 테스트_케이스_3() {
        ListNode list1 = ListNode.builder().add(9).add(9).add(9).add(9).add(9).add(9).add(9).build();
        ListNode list2 = ListNode.builder().add(9).add(9).add(9).add(9).build();

        ListNode expected = ListNode.builder().add(8).add(9).add(9).add(9).add(0).add(0).add(0).add(1).build();
        ListNode result = solution.addTwoNumbers(list1, list2);

        assertThat(expected, is(equalTo(result)));
    }
}

File: leetcode/src/test/kotlin/problems/no2/SolutionTest.kt
package problems.no2

import org.junit.jupiter.api.Assertions.*

import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat
import org.junit.jupiter.api.Test

internal class SolutionTest {
    private val solution: Solution = Solution();

    @Test
    fun 테스트1() {
        val nums = intArrayOf(2, 7, 11, 15)
        val target = 9
        assertThat(intArrayOf(0, 1),`is`(equalTo(solution.twoSum(nums, target))))
    }

    @Test
    fun 테스트2() {
        val nums = intArrayOf(3, 2, 4)
        val target = 6
        assertThat(intArrayOf(1, 2),`is`(equalTo(solution.twoSum(nums, target))))
    }

    @Test
    fun 테스트3() {
        val nums = intArrayOf(3, 3)
        val target = 6
        assertThat(intArrayOf(0, 1),`is`(equalTo(solution.twoSum(nums, target))))
    }
}